src/TestLogger/Core/InputSanitizerJson.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Spekt Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Spekt.TestLogger.Core
+{
+    using System.Text;
+
+    public class InputSanitizerJson : IInputSanitizer
+    {
+        private static readonly char[] EscapeTable;
+        private static readonly char[] EscapeCharacters = { '"', '\\', '\b', '\f', '\n', '\r', '\t' };
+
+        static InputSanitizerJson()
+        {
+            EscapeTable = new char[93];
+            EscapeTable['"'] = '"';
+            EscapeTable['\\'] = '\\';
+            EscapeTable['\b'] = 'b';
+            EscapeTable['\f'] = 'f';
+            EscapeTable['\n'] = 'n';
+            EscapeTable['\r'] = 'r';
+            EscapeTable['\t'] = 't';
+        }
+
+        public string Sanitize(string input)
+        {
+            var sb = new StringBuilder();
```

Comment:
The output string would have nearly the input string length so you should pass it in the ctor to preallocate the memory. Also, you should create the builder after the "happy path".

Reply:
Fair point. We only use the json sanitizer in unit tests so I moved this to that project and added a note on performance. bc8f7a375bab526a045fbd1f510a9f03b5ca1815

btw, what brought you to this PR?

Reply:
(From https://github.com/spekt/junit.testlogger/issues/59)